packages/pn-commons/src/utils/notification.utility.ts

```diff
@@ -930,3 +945,28 @@ export function parseNotificationDetail(
   /* eslint-enable functional/no-let */
   return parsedNotification;
 }
+
+/**
+ * Insert cancelled status n timeline.
+ * @param  {NotificationDetail} notificationDetail
+ */
+const insertCancelledStatusInTimeline = (notificationDetail: NotificationDetail) => {
+  const timelineCancelledElement = notificationDetail.timeline.find(
+    (el) => el.category === TimelineCategory.NOTIFICATION_CANCELLED
+  );
+
+  notificationDetail.timeline.forEach((element) => {
```

Comment:
We can optimize this loop with this two steps:
- before loop check if timelineCancelledElement exists. if exists, DON'T do the loop
- instead a forEach, use find to get the NOTIFICATION_CANCELLATION_REQUEST and after push the new element

Reply:
do it